src/neo/SmartContract/Native/Oracle/OracleContract.cs

```diff
@@ -0,0 +1,224 @@
+using Neo.Cryptography.ECC;
+using Neo.IO;
+using Neo.Ledger;
+using Neo.Network.P2P.Payloads;
+using Neo.Persistence;
+using Neo.SmartContract.Manifest;
+using Neo.SmartContract.Native.Tokens;
+using Neo.VM.Types;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Array = Neo.VM.Types.Array;
+
+namespace Neo.SmartContract.Native
+{
+    public class OracleContract : NativeContract
+    {
+        public override string Name => "Oracle";
+        public override int Id => -5;
+
+        private const byte Prefix_Validator = 37;
+        private const byte Prefix_RequestBaseFee = 13;
+        private const byte Prefix_RequestMaxValidHeight = 33;
+        private const byte Prefix_Request = 21;
+        private const byte Prefix_Response = 27;
+
+        private const long ResponseTxMinFee = 1000;
+        private string[] SupportedProtocol = new string[] { "http", "https" };
+
+        public OracleContract()
+        {
+            Manifest.Features = ContractFeatures.HasStorage | ContractFeatures.Payable;
+            var events = new List<ContractEventDescriptor>(Manifest.Abi.Events)
+            {
+                new ContractEventDescriptor()
+                {
+                    Name = "Request",
+                    Parameters = new ContractParameterDefinition[]
+                    {
+                        new ContractParameterDefinition()
+                        {
+                            Name = "request",
+                            Type = ContractParameterType.InteropInterface
+                        }
+                    }
+                }
+            };
+            Manifest.Abi.Events = events.ToArray();
+        }
+
+        [ContractMethod(0_01000000, CallFlags.AllowStates)]
+        public bool Verify(ApplicationEngine engine)
+        {
+            UInt160 oracleAddress = GetOracleMultiSigAddress(engine.Snapshot);
+            return engine.CheckWitnessInternal(oracleAddress);
+        }
+
+        [ContractMethod(0_01000000, CallFlags.AllowStates)]
+        public bool SetOracleValidators(ApplicationEngine engine, byte[] data)
+        {
+            ECPoint[] validators = data.AsSerializableArray<ECPoint>();
+            UInt160 committeeAddress = NEO.GetCommitteeAddress(engine.Snapshot);
```

Comment:
All the `21` from the committee will be default oracles?

As I mentioned on Erik's PR, it would be great if soon we could have the possibility of registering alternative Oracle Groups.
The Default of any `TX` can still be the Commiteee, but there could be a field for choosing any other service. This looks quite crucial in my opinion.

Reply:
`committeeAddress ` is used to check witness and committee  will not be default oracles.Oracle nodes is designated by committee.